### PR TITLE
feat: or/and conditional filters

### PR DESCRIPTION
**Pull request by yildirayunlu**
Support or/and conditional filters.

### CODE REVIEW

File: packages/nestjsx-crud/test/getList/index.mock.ts [hunk 1]
```diff
@@ -2312,3 +2310,151 @@ nock("https://api.nestjsx-crud.refine.dev:443", { encodedQueryParams: true })
...
+    .query({
+        s: "%7B%22%24and%22%3A%5B%7B%22%24or%22%3A%5B%7B%22%24and%22%3A%5B%7B%22title%22%3A%7B%22%24startsL%22%3A%22a%22%7D%7D%2C%7B%22title%22%3A%7B%22%24contL%22%3A%22heuristic%22%7D%7D%5D%7D%2C%7B%22%24and%22%3A%5B%7B%22title%22%3A%7B%22%24startsL%22%3A%22e%22%7D%7D%2C%7B%22title%22%3A%7B%22%24contL%22%3A%22invoice%22%7D%7D%5D%7D%5D%7D%5D%7D",
+        limit: "10",
+        page: "1",
+        offset: "0",
+    })
+    .reply(
+        200,
+        {
+            data: [
```
Comment (sevkioruc) [inline packages/nestjsx-crud/test/getList/index.mock.ts:2327]: Hey @yildirayunlu,
What do you think about move these test datas to another file and importing here?

Comment (yildirayunlu) [reply packages/nestjsx-crud/test/getList/index.mock.ts:2327]: Hey @sevkioruc, We keep all our mock data in `*.mock.ts` files. Also, we do not use this data in any other test. I don't think there is a need to move it.

File: packages/supabase/src/index.ts [hunk 2]
```diff
@@ -121,6 +137,8 @@ const dataProvider = (
                 .select(metaData?.select ?? "*", {
                     count: "exact",
                 });
+
+            query.eq;
```
Comment (omeraplak) [inline packages/supabase/src/index.ts:141]:
```suggestion
```

File: packages/core/src/definitions/table/index.ts [hunk 3]
```diff
@@ -72,15 +85,32 @@ export const unionFilters = (
...
     prevFilters: CrudFilters = [],
-): CrudFilters =>
-    unionWith(permanentFilter, newFilters, prevFilters, compareFilters).filter(
+): CrudFilters => {
+    const isKeyRequired = newFilters.filter(
+        (f) => (f.operator === "or" || f.operator === "and") && !f.key,
+    );
+
+    if (isKeyRequired.length > 1) {
+        throw new Error("Only one filter without key can be used!");
```
Comment (omeraplak) [inline packages/core/src/definitions/table/index.ts]: We just have to give a warning here

File: packages/core/src/definitions/table/index.ts [hunk 4]
```diff
@@ -72,15 +86,32 @@ export const unionFilters = (
...
     prevFilters: CrudFilters = [],
-): CrudFilters =>
-    unionWith(permanentFilter, newFilters, prevFilters, compareFilters).filter(
+): CrudFilters => {
+    const isKeyRequired = newFilters.filter(
+        (f) => (f.operator === "or" || f.operator === "and") && !f.key,
+    );
+
+    if (isKeyRequired.length > 1) {
+        warnOnce(true, `Only one filter without key can be used!`);
```
Comment (omeraplak) [inline packages/core/src/definitions/table/index.ts:95]: Let's add some more details to this error and direct it to the part in the document where there is an explanation of how to use it.

File: packages/core/src/definitions/table/index.ts (hunk 3 above)
Comment (yildirayunlu) [reply packages/core/src/definitions/table/index.ts]: fix [this commit](https://github.com/pankod/refine/pull/2751/commits/71ab469e84a254c60cbd895c3fa10f9e82575fee

File: packages/core/src/definitions/table/index.ts [hunk 5]
```diff
@@ -72,15 +86,35 @@ export const unionFilters = (
...
-    unionWith(permanentFilter, newFilters, prevFilters, compareFilters).filter(
+): CrudFilters => {
+    const isKeyRequired = newFilters.filter(
+        (f) => (f.operator === "or" || f.operator === "and") && !f.key,
+    );
+
+    if (isKeyRequired.length > 1) {
+        warnOnce(
+            true,
+            `[conditionalFilters]: You have created multiple Conditional Filters at the top level, this requires the key parameter. \nFor more information, see docs/advanced-tutorials/data-provider/handling-filters/#top-level-multiple-conditional-filters-usage`,
```
Comment (omeraplak) [inline packages/core/src/definitions/table/index.ts:97]:
```suggestion
            `[conditionalFilters]: You have created multiple Conditional Filters at the top level, this requires the key parameter. \nFor more information, see https://refine.dev/docs/advanced-tutorials/data-provider/handling-filters/#top-level-multiple-conditional-filters-usage`,
```

**Review by omeraplak — APPROVED** [review]

**Pull request merged**